staging/src/k8s.io/client-go/testing/fixture.go

```diff
@@ -144,32 +129,35 @@ func ObjectReaction(tracker ObjectTracker, mapper meta.RESTMapper) ReactionFunc
 }
 
 type tracker struct {
-	registry *registered.APIRegistrationManager
-	scheme   ObjectScheme
-	decoder  runtime.Decoder
-	lock     sync.RWMutex
-	objects  map[schema.GroupVersionKind][]runtime.Object
+	scheme  ObjectScheme
+	decoder runtime.Decoder
+	lock    sync.RWMutex
+	objects map[schema.GroupVersionResource][]runtime.Object
 }
 
 var _ ObjectTracker = &tracker{}
 
 // NewObjectTracker returns an ObjectTracker that can be used to keep track
 // of objects for the fake clientset. Mostly useful for unit tests.
-func NewObjectTracker(registry *registered.APIRegistrationManager, scheme ObjectScheme, decoder runtime.Decoder) ObjectTracker {
+func NewObjectTracker(scheme ObjectScheme, decoder runtime.Decoder) ObjectTracker {
 	return &tracker{
-		registry: registry,
-		scheme:   scheme,
-		decoder:  decoder,
-		objects:  make(map[schema.GroupVersionKind][]runtime.Object),
+		scheme:  scheme,
+		decoder: decoder,
+		objects: make(map[schema.GroupVersionResource][]runtime.Object),
 	}
 }
 
-func (t *tracker) List(gvk schema.GroupVersionKind, ns string) (runtime.Object, error) {
+func (t *tracker) List(gvr schema.GroupVersionResource, gvk schema.GroupVersionKind, ns string) (runtime.Object, error) {
 	// Heuristic for list kind: original kind + List suffix. Might
 	// not always be true but this tracker has a pretty limited
 	// understanding of the actual API model.
 	listGVK := gvk
 	listGVK.Kind = listGVK.Kind + "List"
+	// GVK does have the concept of "internal version". The scheme recognizes
+	// the runtime.APIVersionInternal, but not the empty string.
+	if listGVK.Version == "" {
```

Comment:
Mostly in the generated fake clients, like this [line](https://github.com/kubernetes/kubernetes/pull/45159/files#diff-1ce5095abadb94bcbff3015537351389R37). These are hidden from unit test writers so i think we should go with the approach that causes minimal changes.